spec/requests/api/submissions/index_spec.rb

```diff
@@ -31,6 +31,22 @@
       expect(body).to eq []
     end
 
+    it "returns only draft submissions when state is draft" do
+      Fabricate(:submission, user: user, state: "draft")
+      Fabricate(:submission, user: user, state: "approved")
+      Fabricate(:submission, user: user, state: "submitted")
+
+      get "/api/submissions",
+        params: {
+          state: ["draft", "submitted"]
+        },
+        headers: headers
+
+      expect(response.status).to eq 200
+      body = JSON.parse(response.body)
+      expect(body.length).to eq 2
```

Comment:
Maybe we can add a test to check if the correct submissions are returned. 

```suggestion
      expect(body.length).to eq 2
      expect(body.map { |s| s.state ).to eq ["draft", "submitted"]
```

Reply:
good thought!